frontend/src/components/pages/CourseDashboard.js

```diff
@@ -22,10 +67,44 @@ export function CourseDashboard() {
             })
             .then((data) => setCourseInfo(data.courses))
             .catch((error) => setError(error.message));
+        }
+
+        async function fetchUser() {
+            await fetch(`http://localhost:4000/user/${userId}`)
+            .then((response) => {
+                if (!response.ok) {
+                    throw new Error('Network response was not ok');
+                }
+                return response.json();
+            })
+            .then((data) => setUserInfo(data.user))
+            .catch((error) => setError(error.message));
+        }
+
+        fetchCourses();
+        fetchUser();
     }, []);
 
     if (error) return <p>Error: {error}</p>;
-    if (!courseInfo) return <p>Loading...</p>;
+    if (!courseInfo || !userInfo) return <p>Loading...</p>;
+
+    let createButton = null;
+    if (userInfo.role === "educator"){
+        createButton = <Modal
+            title={"Create New Course"}
+            trigger={
+                <div className="bg-blue-500 p-2 rounded shadow hover:bg-blue-700 m-auto text-center text-sm text-white font-semibold hover:cursor-pointer" > 
+                    <div>+ New Course</div> 
+                </div>
+            }
+            inputFields={{
+                title : "Course Name",
+                description : "Course Description"
+            }}
+            changeHandler={handleChange}
+            confirmHandler={handleCreateCourse}
+        />
+    }
 
     var courseList = [];
```

Comment:
nit: I didn't see that in the previous PR, but we want to truncate the title if it is too long. Maybe just add ``truncate overflow-hidden`` in the course.title className.